### ISSUE TITLE
Failed in buildding project

### ISSUE THREAD

**Issue by Renhy** [nokiatech/heif#56]
I use these commands for build this project:
```
cd heif/build   
cmake3 ../srcs   
camke3 --build .
```
and I'm using
gcc 4.8.5 and cmake 3.11.2 on CentOS 7

but I failed with following error, somebody help me please?
```
/root/heif/heif/srcs/api-cpp/Heif.cpp:47:43: error: redeclaration ‘HEIFPP::Heif::InvalidGroup’ differs in ‘constexpr’
 constexpr HEIF::GroupId Heif::InvalidGroup((std::uint32_t) 0);
                                           ^
In file included from /root/heif/heif/srcs/api-cpp/DecoderConfiguration.h:15:0,
                 from /root/heif/heif/srcs/api-cpp/AACDecoderConfiguration.h:15,
                 from /root/heif/heif/srcs/api-cpp/Heif.cpp:16:
/root/heif/heif/srcs/api-cpp/Heif.h:77:36: error: from previous declaration ‘HEIFPP::Heif::InvalidGroup’
         static const HEIF::GroupId InvalidGroup;
                                    ^
/root/heif/heif/srcs/api-cpp/Heif.cpp:47:43: error: ‘HEIFPP::Heif::InvalidGroup’ declared ‘constexpr’ outside its class
 constexpr HEIF::GroupId Heif::InvalidGroup((std::uint32_t) 0);
                                           ^
/root/heif/heif/srcs/api-cpp/Heif.cpp:47:43: error: declaration of ‘const HEIF::GroupId HEIFPP::Heif::InvalidGroup’ outside of class is not definition [-fpermissive]
/root/heif/heif/srcs/api-cpp/Heif.cpp:48:42: error: redeclaration ‘HEIFPP::Heif::InvalidItem’ differs in ‘constexpr’
 constexpr HEIF::ImageId Heif::InvalidItem((std::uint32_t) 0);
                                          ^
In file included from /root/heif/heif/srcs/api-cpp/DecoderConfiguration.h:15:0,
                 from /root/heif/heif/srcs/api-cpp/AACDecoderConfiguration.h:15,
                 from /root/heif/heif/srcs/api-cpp/Heif.cpp:16:
/root/heif/heif/srcs/api-cpp/Heif.h:78:36: error: from previous declaration ‘HEIFPP::Heif::InvalidItem’
         static const HEIF::ImageId InvalidItem;
                                    ^
/root/heif/heif/srcs/api-cpp/Heif.cpp:48:42: error: ‘HEIFPP::Heif::InvalidItem’ declared ‘constexpr’ outside its class
 constexpr HEIF::ImageId Heif::InvalidItem((std::uint32_t) 0);
                                          ^
/root/heif/heif/srcs/api-cpp/Heif.cpp:48:42: error: declaration of ‘const HEIF::ImageId HEIFPP::Heif::InvalidItem’ outside of class is not definition [-fpermissive]
/root/heif/heif/srcs/api-cpp/Heif.cpp:49:49: error: redeclaration ‘HEIFPP::Heif::InvalidSequence’ differs in ‘constexpr’
 constexpr HEIF::SequenceId Heif::InvalidSequence((std::uint32_t) ~0);
                                                 ^
In file included from /root/heif/heif/srcs/api-cpp/DecoderConfiguration.h:15:0,
                 from /root/heif/heif/srcs/api-cpp/AACDecoderConfiguration.h:15,
                 from /root/heif/heif/srcs/api-cpp/Heif.cpp:16:
/root/heif/heif/srcs/api-cpp/Heif.h:79:39: error: from previous declaration ‘HEIFPP::Heif::InvalidSequence’
         static const HEIF::SequenceId InvalidSequence;
                                       ^
/root/heif/heif/srcs/api-cpp/Heif.cpp:49:49: error: ‘HEIFPP::Heif::InvalidSequence’ declared ‘constexpr’ outside its class
 constexpr HEIF::SequenceId Heif::InvalidSequence((std::uint32_t) ~0);
                                                 ^
/root/heif/heif/srcs/api-cpp/Heif.cpp:49:49: error: declaration of ‘const HEIF::SequenceId HEIFPP::Heif::InvalidSequence’ outside of class is not definition [-fpermissive]
/root/heif/heif/srcs/api-cpp/Heif.cpp:50:59: error: redeclaration ‘HEIFPP::Heif::InvalidSequenceImage’ differs in ‘constexpr’
 constexpr HEIF::SequenceImageId Heif::InvalidSequenceImage((std::uint32_t) ~0);
                                                           ^
In file included from /root/heif/heif/srcs/api-cpp/DecoderConfiguration.h:15:0,
                 from /root/heif/heif/srcs/api-cpp/AACDecoderConfiguration.h:15,
                 from /root/heif/heif/srcs/api-cpp/Heif.cpp:16:
/root/heif/heif/srcs/api-cpp/Heif.h:80:44: error: from previous declaration ‘HEIFPP::Heif::InvalidSequenceImage’
         static const HEIF::SequenceImageId InvalidSequenceImage;
                                            ^
/root/heif/heif/srcs/api-cpp/Heif.cpp:50:59: error: ‘HEIFPP::Heif::InvalidSequenceImage’ declared ‘constexpr’ outside its class
 constexpr HEIF::SequenceImageId Heif::InvalidSequenceImage((std::uint32_t) ~0);
                                                           ^
/root/heif/heif/srcs/api-cpp/Heif.cpp:50:59: error: declaration of ‘const HEIF::SequenceImageId HEIFPP::Heif::InvalidSequenceImage’ outside of class is not definition [-fpermissive]
/root/heif/heif/srcs/api-cpp/Heif.cpp:51:49: error: redeclaration ‘HEIFPP::Heif::InvalidProperty’ differs in ‘constexpr’
 constexpr HEIF::PropertyId Heif::InvalidProperty((std::uint32_t) 0);
                                                 ^
In file included from /root/heif/heif/srcs/api-cpp/DecoderConfiguration.h:15:0,
                 from /root/heif/heif/srcs/api-cpp/AACDecoderConfiguration.h:15,
                 from /root/heif/heif/srcs/api-cpp/Heif.cpp:16:
/root/heif/heif/srcs/api-cpp/Heif.h:81:39: error: from previous declaration ‘HEIFPP::Heif::InvalidProperty’
         static const HEIF::PropertyId InvalidProperty;
                                       ^
/root/heif/heif/srcs/api-cpp/Heif.cpp:51:49: error: ‘HEIFPP::Heif::InvalidProperty’ declared ‘constexpr’ outside its class
 constexpr HEIF::PropertyId Heif::InvalidProperty((std::uint32_t) 0);
                                                 ^
/root/heif/heif/srcs/api-cpp/Heif.cpp:51:49: error: declaration of ‘const HEIF::PropertyId HEIFPP::Heif::InvalidProperty’ outside of class is not definition [-fpermissive]
/root/heif/heif/srcs/api-cpp/Heif.cpp:52:59: error: redeclaration ‘HEIFPP::Heif::InvalidDecoderConfig’ differs in ‘constexpr’
 constexpr HEIF::DecoderConfigId Heif::InvalidDecoderConfig((std::uint32_t) 0);
                                                           ^
In file included from /root/heif/heif/srcs/api-cpp/DecoderConfiguration.h:15:0,
                 from /root/heif/heif/srcs/api-cpp/AACDecoderConfiguration.h:15,
                 from /root/heif/heif/srcs/api-cpp/Heif.cpp:16:
/root/heif/heif/srcs/api-cpp/Heif.h:82:44: error: from previous declaration ‘HEIFPP::Heif::InvalidDecoderConfig’
         static const HEIF::DecoderConfigId InvalidDecoderConfig;
                                            ^
/root/heif/heif/srcs/api-cpp/Heif.cpp:52:59: error: ‘HEIFPP::Heif::InvalidDecoderConfig’ declared ‘constexpr’ outside its class
 constexpr HEIF::DecoderConfigId Heif::InvalidDecoderConfig((std::uint32_t) 0);
                                                           ^
/root/heif/heif/srcs/api-cpp/Heif.cpp:52:59: error: declaration of ‘const HEIF::DecoderConfigId HEIFPP::Heif::InvalidDecoderConfig’ outside of class is not definition [-fpermissive]
/root/heif/heif/srcs/api-cpp/Heif.cpp:53:51: error: redeclaration ‘HEIFPP::Heif::InvalidMediaData’ differs in ‘constexpr’
 constexpr HEIF::MediaDataId Heif::InvalidMediaData((std::uint32_t) 0);
                                                   ^
In file included from /root/heif/heif/srcs/api-cpp/DecoderConfiguration.h:15:0,
                 from /root/heif/heif/srcs/api-cpp/AACDecoderConfiguration.h:15,
                 from /root/heif/heif/srcs/api-cpp/Heif.cpp:16:
/root/heif/heif/srcs/api-cpp/Heif.h:83:40: error: from previous declaration ‘HEIFPP::Heif::InvalidMediaData’
         static const HEIF::MediaDataId InvalidMediaData;
                                        ^
/root/heif/heif/srcs/api-cpp/Heif.cpp:53:51: error: ‘HEIFPP::Heif::InvalidMediaData’ declared ‘constexpr’ outside its class
 constexpr HEIF::MediaDataId Heif::InvalidMediaData((std::uint32_t) 0);
                                                   ^
/root/heif/heif/srcs/api-cpp/Heif.cpp:53:51: error: declaration of ‘const HEIF::MediaDataId HEIFPP::Heif::InvalidMediaData’ outside of class is not definition [-fpermissive]
make[2]: *** [api-cpp/CMakeFiles/heifpp.dir/Heif.cpp.o] Error 1
make[2]: Leaving directory `/root/heif/heif/build'
make[1]: *** [api-cpp/CMakeFiles/heifpp.dir/all] Error 2
make[1]: Leaving directory `/root/heif/heif/build'
make: *** [all] Error 2

```

**Comment by nokiatech** [comment]
Hi,
Please pull v3.3.1 - it will fix compilation with older gcc v4.8.5.
Br,

**Comment by Renhy** [comment]
@nokiatech It works! Thanks
